internal/vm/builder.go

```diff
@@ -0,0 +1,83 @@
+package vm
+
+import (
+	"context"
+)
+
+type UVMBuilder interface {
+	// Create will create the Utility VM in a paused/powered off state with whatever is present in the implementation
+	// of the interfaces config at the time of the call.
+	Create(ctx context.Context) (UVM, error)
+
+	SCSIManager
```

Comment:
Putting the interfaces here has the same problem with putting them on `UVM`, no?

Reply:
That's what I'm fighting with here.  With HCS it has everything we need, so the HCS `UVM` should implement every interface, no worries here. 

For remotevm (just the name I have for the `UVM` impl that uses the ttrpc definitions), we don't know what's supported or not, so we have to rely on the `Supported` call also in addition to the typecast. Ideally when the whole ttrpc surface has everything that we care we'd also implement every interface for remotevm, or truthfully even right now it could and just stub the methods that don't have generated code yet (plan9 and friends) with `ErrNotSupported`, so I personally don't find it too awful to actually implement everything. 

If we're in a world where the proto has matched HCS entirely, and these interfaces are embedded in the builder (or just on UVM itself), then we'd have every method filled out for HCS, and every method filled out for remotevm and then it would just be up to the caller to check `uvm.Supported(vm.Foo, vm.Bar)` before things. 

Are you saying you'd prefer to do the typecast dance + Supported() for creating the initial document too? This brings to mind a hole in the ttrpc `CapabilitiesVM` response in that we only return whether Add/Remove/Update is supported and not just whether it's supported at creation time.